src/components/common/BaseButton/BaseButton.interface.ts

```diff
@@ -0,0 +1,14 @@
+type TClassName = 'primary' | 'support' | 'support-line' | 'outline' | 'tag';
+type TType = 'button' | 'submit' | 'reset';
+
+interface IButtonProps {
```

Comment:
```suggestion
interface IButtonProps extends HTMLButtonElement{
```
으로 직접 상속해서 사용하면 하나하나 적용하는 휴먼 에러를 막을수 있을거 같은데 고려해보셔도 좋을듯합니다!